hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieDefaultTimeline.java

```diff
@@ -540,10 +542,10 @@ private Option<HoodieInstant> getLastOrFirstPendingClusterInstant(boolean isLast
 
   @Override
   public boolean isPendingClusterInstant(String instantTime) {
-    HoodieTimeline potentialTimeline = getCommitsTimeline().filterPendingReplaceTimeline().filter(i -> i.getTimestamp().equals(instantTime));
-    if (potentialTimeline.countInstants() == 0) {
+    if (!getOrCreatePendingReplaceInstantSet().contains(instantTime)) {
       return false;
     }
+    HoodieTimeline potentialTimeline = getCommitsTimeline().filterPendingReplaceTimeline().filter(i -> i.getTimestamp().equals(instantTime));
     if (potentialTimeline.countInstants() > 1) {
       throw new IllegalStateException("Multiple instants with same timestamp: " + potentialTimeline);
```

Comment:
Maybe we can store the cache as `instant time -> duplicate instant time count`, so that we can always have a `O1` query for the check.

And to be clear, the cache should only cache pending clustering instant, so `ClusteringUtils.isClusteringInstant` should be used in the filter of `getOrCreatePendingReplaceInstantSet`.

Reply:
What do you mean by duplicate?

Reply:
When multiple same instant time instant are on the timeline: `potentialTimeline.countInstants() > 1`, this check is slow, let's improve it.